controller/api/public/stat_summary.go

```diff
@@ -88,13 +88,23 @@ func (s *grpcServer) StatSummary(ctx context.Context, req *pb.StatSummaryRequest
 		statReq := &pb.StatSummaryRequest{
 			Selector: &pb.ResourceSelection{
 				Resource: &pb.Resource{
-					Type:      resource,
 					Namespace: req.Selector.Resource.Namespace,
+					Type:      resource,
+					Name:      req.Selector.Resource.Name,
 				},
 				LabelSelector: req.Selector.LabelSelector,
 			},
 			TimeWindow: req.TimeWindow,
 		}
+		if req.GetFromResource() != nil {
+			statReq.Outbound = &pb.StatSummaryRequest_FromResource{
+				FromResource: req.GetFromResource(),
+			}
+		} else if req.GetToResource() != nil {
+			statReq.Outbound = &pb.StatSummaryRequest_ToResource{
+				ToResource: req.GetToResource(),
+			}
+		}
```

Comment:
Rather than copying over each field individually and needing to update this each time we add new fields, I think it would be better to clone the req message using [protobuf's `Clone` func](https://github.com/golang/protobuf/blob/master/proto/clone.go#L44). Something like:

```
	for _, resource := range resourcesToQuery {
		statReq := proto.Clone(req).(*pb.StatSummaryRequest)
		statReq.Selector.Resource.Type = resource
		go func() {
			resultChan <- s.resourceQuery(ctx, statReq)
		}()
	}
```

Reply:
Wow ok, this implementation is better.

Reply:
whoa! yeah, `Clone` is great.